plasmapy/particles/particle_class.py

```diff
@@ -96,6 +98,38 @@ def _category_errmsg(particle, category: str) -> str:
     return errmsg
 
 
+class ParticleJSONDecoder(json.JSONDecoder):
+    def __init__(self, *args, **kwargs):
+        json.JSONDecoder.__init__(self, object_hook=self.object_hook, *args, **kwargs)
+
+    def object_hook(self, json_dict):
```

Comment:
```suggestion
    def __init__(self, *, object_hook=None, parse_float=None,
                 parse_int=None, parse_constant=None, strict=True,
                 object_pairs_hook=None):

        if object_hook is None:
            object_hook = self.particle_hook

        json.JSONDecoder.__init__(self,
                                  object_hook=object_hook,
                                  parse_float=parse_float,
                                  parse_int=parse_int,
                                  parse_constant=parse_constant,
                                  strict=strict,
                                  object_pairs_hook=object_pairs_hook)

    @staticmethod
    def particle_hook(json_dict):
```

This structure is more friendly with the internal code of `JSONDecode`.   Not so simply put, `JSONDecode` sets the `object_hook` method from the equally named input parameter (i.e. `self.object_hook = object_hook`).  If we define our hook method as `object_hook` then the `JSONDecode` statement will always be overridden by our method.